### PR TITLE
Feature/add unit tests

### PR DESCRIPTION
**Pull request by frostblogNet**
Please make sure these boxes are checked before submitting your PR - thank you!

- [ x] Pull request is based against develop branch
- [x ] README.md reflects changes (if applicable)
- [x ] New files contain a license header

### Issue

This PR is adding some unit tests

### CODE REVIEW

File: Test/Unit/Block/Price/DetailsTest.php [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    }
+
+    public function testSetSaleableItem(): void
+    {
+        $this->sut->setData('tax_rate', '19');
+        $saleableItemMock = $this->getMockBuilder(SaleableInterface::class)
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+        $this->sut->setSaleableItem($saleableItemMock);
+        $this->assertSame(null, $this->sut->getData('tax_rate'));
```
Comment (sprankhub) [inline Test/Unit/Block/Price/DetailsTest.php:98]: Maybe use `assertNull`?

File: Test/Unit/Block/Price/DetailsTest.php [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+            ->method('getRateRequest')
+            ->willReturn($dataMock);
+
+        $this->taxCalculationMock
+            ->expects($this->at(1))
+            ->method('getRate')
+            ->willReturn(5);
+
+        $this->sut->setSaleableItem($saleableItem2Mock);
+        $this->assertInstanceOf(Phrase::class, $this->sut->getFormattedTaxRate());
```
Comment (sprankhub) [inline Test/Unit/Block/Price/DetailsTest.php:182]: Should we add a more concrete assertion (applies to all `$this->assertInstanceOf(Phrase::class, ...)` cases)?

File: Test/Unit/Block/Price/DetailsTest.php (hunk 1 above)
Comment (frostblogNet) [reply Test/Unit/Block/Price/DetailsTest.php]: I changed this to assertNull

File: Test/Unit/Block/Price/DetailsTest.php (hunk 2 above)
Comment (frostblogNet) [reply Test/Unit/Block/Price/DetailsTest.php]: Yes you are right. I have changed it.

**Review by sprankhub — APPROVED** [review]

**Pull request merged**